### ISSUE TITLE
Allow Double values to be passed via React Native bridge (as opposed to forcing double)

### ISSUE THREAD

**Issue by colmsnowplow** [facebook/react-native#29054]
## Description

Passing a null value is acceptable for Double type in Java, but not allowed for double. 

Javascript and obj-c have no concept of any such limitation. 

The React-Native bridge's [getDouble()](https://github.com/facebook/react-native/blob/5c474ac24c86857ad2ff9d2444d3b82e171a7cee/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableMap.java#L28) method forces one to use `double` type.

Since it's very easy to think of use cases where a Double value should be allowed to be null, and values which are compatible with the js and obj-c layer are not compatible with the Java layer, this seems to me to be a bug.

## React Native version:
All versions

## Steps To Reproduce

Attempt to pass a null value to the `getDouble()` method.

## Expected Results
Null value is allowed and handled as a Double rather than a double - or at least a Double-compatible method can be used.

**Comment by safaiyeh** [comment]
Hi @colmsnowplow thanks for the issue. This seems to be a specific optimization, have you encountered a use case where this caused you problems?

**Comment by colmsnowplow** [comment]
Yeah - I worked around it [here](https://github.com/snowplow-incubator/snowplow-react-native-tracker/blob/6eb35fd52b119196d8d29f5b93f8aaa9ef78e126/android/src/main/java/com/snowplowanalytics/react/util/EventUtil.java#L66-L68) by ignoring the value if it's null or empty. But this isn't the desired behaviour.

In this instance it's just an irritation that it's more effort than it's worth to elicit the exact behaviour I want. 

In the general case, the framework involves passing values from Javascript to Java. Javascript allows nullability. Java has a type which was specifically created to handle nullable cases. It's pretty commonplace that a value could be a nullable Double. I'm not saying it's a bug or a showstopper or that there's no other way. I'm saying it's a pain, and for me there's a good case for making a devs life easier here by providing a method to easily grab a Double.

**Comment by safaiyeh** [comment]
Appreciate the link to the code snippet! Tagged Issue Manager Attention for the team to take a look

**Comment by chrisglein** [comment]
I know we've had lots of discussions on the Windows side about how to handle doubles in the native code. I'm curious if @vmoroz can weigh in with what we did and/or if we have any suggestions on changes to core that we'd like to see and/or how this is affected by the Turbo Modules world (also pinging @stmoy in case he has that info).
